modules/system/console/PluginRollback.php

```diff
@@ -32,24 +34,37 @@ class PluginRollback extends Command
      */
     public function handle()
     {
-        if ($this->option('force') || $this->confirm('Do you wish to continue? This option is nonreversible. [yes|no]')) {
-            /*
+        /*
             * Lookup plugin
             */
-            $pluginName = $this->argument('name');
-            $pluginName = PluginManager::instance()->normalizeIdentifier($pluginName);
-            if (!PluginManager::instance()->exists($pluginName)) {
-                throw new \InvalidArgumentException(sprintf('Plugin "%s" not found.', $pluginName));
+        $pluginName = $this->argument('name');
+        $pluginName = PluginManager::instance()->normalizeIdentifier($pluginName);
+        if (!PluginManager::instance()->exists($pluginName)) {
+            throw new \InvalidArgumentException(sprintf('Plugin "%s" not found.', $pluginName));
+        }
+
+        $stopOnVersion = ltrim(($this->argument('version') ?: null), 'v');
+
+        if ($stopOnVersion) {
+            if (!VersionManager::instance()->hasDatabaseVersion($pluginName, $stopOnVersion)) {
+                throw new \InvalidArgumentException(Lang::get('system::lang.updates.plugin_version_not_found'));
             }
+            $confirmQuestion = 'Do you want to continue reverting to version ' . $stopOnVersion . '? This option is nonreversible. [yes|no]';
+        } else {
+            $confirmQuestion = 'Do you wish to continue reverting all versions? This option is nonreversible. [yes|no]';
```

Comment:
@prhost 
```suggestion
            $confirmQuestion = 'Please confirm that you wish to completely rollback this plugin. This may result in potential data loss.';
```